receiver/statsdreceiver/internal/transport/uds_server.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package transport // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/internal/transport"
+
+import (
+	"fmt"
+	"net"
+	"os"
+)
+
+type udsServer struct {
+	packetServer
+}
+
+// Ensure that Server is implemented on UDS Server.
+var _ (Server) = (*udsServer)(nil)
+
+// NewUDSServer creates a transport.Server using Unixgram as its transport.
+func NewUDSServer(transport Transport, socketPath string) (Server, error) {
+	if !transport.IsPacketTransport() {
+		return nil, fmt.Errorf("NewUDSServer with %s: %w", transport.String(), ErrUnsupportedPacketTransport)
+	}
+
+	if err := prepareSocket(socketPath); err != nil {
+		return nil, err
+	}
+
+	conn, err := net.ListenPacket(transport.String(), socketPath)
+	if err != nil {
+		return nil, fmt.Errorf("starting to listen %s socket: %w", transport.String(), err)
+	}
+
+	return &udsServer{
+		packetServer: packetServer{
+			packetConn: conn,
+			transport:  transport,
+		},
+	}, nil
+}
+
+// Close closes the server.
+func (u *udsServer) Close() error {
+	os.Remove(u.packetConn.LocalAddr().String())
+	return u.packetConn.Close()
+}
+
+func prepareSocket(socketPath string) error {
+	if _, err := os.Stat(socketPath); err == nil {
+		// File exists, remove it
```

Comment:
Wouldn't be safer to just fail here instead of "force"-removing the socket? What if this socket is already in use by another Collector instance or just another application?

Reply:
Hmm that's fair. At my previous company, we deployed the otel collector as a daemonset so we used this pattern in case the socket wasn't properly cleaned up from an old instance of the collector. What are your thoughts in that case?

Reply:
I don't think it should be responsibility of the new collector instance itself to clean this up. If somebody would want to automate this, they could easily hook an init container or any kind of pre-run bash script to take care of it on administrative level.

Having this being part of the Collector would even allow to someone using it maliciously.

Reply:
I agree with @ChrsMark here. An init container can take care of that. I don't think collector should assume write access to the filesystem

Reply:
Can this be a configurable optional behavior?

Reply:
Makes sense. I've removed this check